arrow-optics/src/main/kotlin/arrow/optics/Fold.kt

```diff
@@ -23,13 +10,12 @@ import arrow.typeclasses.Monoid
  * @param S the source of a [Fold]
  * @param A the target of a [Fold]
  */
-@higherkind
-interface Fold<S, A> : FoldOf<S, A> {
+interface Fold<S, A> {
 
   /**
    * Map each target to a type R and use a Monoid to fold the results
    */
-  fun <R> foldMap(M: Monoid<R>, s: S, f: (A) -> R): R
+  fun <R> foldMap(s: S, empty: R, combine: (R, R) -> R, map: (A) -> R): R
```

Comment:
👌

Reply:
Why do we want to get rid of `Monoid`? It's not a HKT typeclass, nor is it a very confusing one? I however would be in favor of keeping this signature next to a `Monoid` one.

Reply:
Not sure we made a decision on Monoid or anyone has yet but Monoid has an allocation cost that Plus and Empty don't since they match structurally with that is in the std lib for example Int::plus and 0. If you ask for Monoid then you instead need to create an object instance to wrap the two methods.

Reply:
Monoid instances have one allocation cost unless you build one per invocation and usually the instances are objects created once.

Having to always fill in the two methods may be just as annoying as always passing an object if the same monoid is used a lot, why not offer both in this case?

Reply:
Fair,but most `Monoid`s can be implemented with `object` unless they rely on another `Monoid`. A concatenating `Monoid` for `List` could be cached, if you don't mind the cast in the summoning function.